extensions/blocks/podcast-episodes/edit.js

```diff
@@ -3,49 +3,139 @@
 /**
  * External dependencies
  */
+import { Component } from '@wordpress/element';
+import {
+	Button,
+	Disabled,
+	ExternalLink,
+	PanelBody,
+	Placeholder,
+	RangeControl,
+	TextControl,
+	Toolbar,
+	Notice,
+} from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { BlockIcon } from '@wordpress/block-editor';
-import { Placeholder, withNotices } from '@wordpress/components';
-import { useState } from '@wordpress/element';
+import { BlockControls, BlockIcon, InspectorControls } from '@wordpress/block-editor';
+import ServerSideRender from '@wordpress/server-side-render';
+import { isURL } from '@wordpress/url';
 
 /**
  * Internal dependencies
  */
 import './editor.scss';
-import icon from './icon';
-
-function PodcastEpisodesEdit( {
-	attributes,
-	className,
-	noticeOperations,
-	noticeUI,
-	setAttributes,
-} ) {
-	/**
-	 * Write the block editor UI.
-	 *
-	 * @returns {object} The UI displayed when user edits this block.
-	 */
-	const [ notice, setNotice ] = useState();
-
-	/* Call this function when you want to show an error in the placeholder. */
-	const setErrorNotice = () => {
-		noticeOperations.removeAllNotices();
-		noticeOperations.createErrorNotice( __( 'Put error message here.', 'jetpack' ) );
+import { edit, queueMusic } from './icons/';
+
+const DEFAULT_MIN_ITEMS = 1;
+const DEFAULT_MAX_ITEMS = 10;
+
+class PodcastEpisodesEdit extends Component {
+	constructor() {
+		super( ...arguments );
+		this.toggleAttribute = this.toggleAttribute.bind( this );
+		this.onSubmitURL = this.onSubmitURL.bind( this );
+
+		this.state = {
+			editing: ! this.props.attributes.url,
+			urlError: '',
+		};
+	}
+
+	toggleAttribute( propName ) {
```

Comment:
is this method being used in any place?
Probably I'm missing something.

Reply:
It was used in the first big PR for some controls in the sidebar. I think we might remove it now and only add it once there is some code using it

Reply:
Removed.